cmd/collector/app/span_processor.go

```diff
@@ -126,10 +128,18 @@ func newSpanProcessor(spanWriter spanstore.Writer, opts ...Option) *spanProcesso
 	return &sp
 }
 
-// Stop halts the span processor and all its go-routines.
+// Stop is an alias to #Close(). Deprecated.
 func (sp *spanProcessor) Stop() {
-	close(sp.stopCh)
-	sp.queue.Stop()
+	sp.Close()
+}
+
+func (sp *spanProcessor) Close() error {
+	if sp.stopped.CAS(false, true) {
```

Comment:
Usually this kind of double-close protection is a code smell for a problem elsewhere (such as bad ownership). Do we really need it?